src/query/src/parser.rs

```diff
@@ -0,0 +1,36 @@
+use common_error::prelude::BoxedError;
+use common_telemetry::timer;
+use promql_parser::parser::EvalStmt;
+use snafu::ResultExt;
+use sql::dialect::GenericDialect;
+use sql::parser::ParserContext;
+use sql::statements::statement::Statement;
+
+use crate::error::{MultipleStatementsSnafu, QueryParseSnafu, Result};
+use crate::metric::METRIC_PARSE_SQL_ELAPSED;
+
+pub enum QueryStatement {
+    SQL(Statement),
+    PromQL(EvalStmt),
```

Comment:
Use `Sql`? Since in Rust we usually only capitalize the first letter, like `Grpc`. But `PromQL` we might need to use `Promql`
```suggestion
    Sql(Statement),
    Promql(EvalStmt),
```

As mentioned in the [style-guide](https://pingcap.github.io/style-guide/rust/naming.html)
> Use acronyms where the acronym is standard, e.g., Sql rather than StandardQueryLanguage. Treat acronyms as words, e.g., GrpcType or grpc_variable, not GRPCType or g_r_p_c_variable/GRPC_variable. In particular, ‘TiKV’ should be capitalised as Tikv, e.g., TikvServer, not TiKVServer Treat contractions of multiple words as one word, e.g., Stdin rather than StdIn.